src/map-extent.js

```diff
@@ -48,58 +62,724 @@ export class MapExtent extends HTMLElement {
+    // doesn't trigger the event handler because it's not user-caused AFAICT
+    this._layerControlCheckbox.checked = this.checked;
+  }
+  _validateLayerControlContainerHidden() {
+    let extentsFieldset = this._layer._layerEl._propertiesGroupAnatomy;
+    let nodeToSearch = this.parentLayer.shadowRoot || this.parentLayer;
+    if (
+      nodeToSearch.querySelectorAll('map-extent:not([hidden])').length === 0
+    ) {
+      extentsFieldset.setAttribute('hidden', '');
+    } else {
+      extentsFieldset.removeAttribute('hidden');
+    }
+  }
+  disconnectedCallback() {
+    if (
+      this.hasAttribute('data-moving') ||
+      this.parentLayer.hasAttribute('data-moving')
+    )
+      return;
+    this._validateLayerControlContainerHidden();
+    // remove layer control for map-extent from layer control DOM
+    this._layerControlHTML.remove();
+
+    this._map.removeLayer(this._templatedLayer);
+    delete this._templatedLayer;
+    delete this.parentLayer.bounds;
+  }
+  _changeOpacity() {
+    if (this.opacity >= 0 && this.opacity <= 1.0) {
+      this._templatedLayer.changeOpacity(this.opacity);
+      this._templateVars.opacity = this.opacity;
+      this._opacitySlider.value = this.opacity;
+    }
+  }
+  _calculateBounds() {
+    // await this.whenReady();
+    let bounds = null,
+      zoomMax = 0,
+      zoomMin = 0,
+      maxNativeZoom = 0,
+      minNativeZoom = 0;
+    if (this.checked) {
+      for (let j = 0; j < this._templateVars.length; j++) {
+        let inputData = M._extractInputBounds(this._templateVars[j]);
+        this._templateVars[j].tempExtentBounds = inputData.bounds;
+        this._templateVars[j].extentZoomBounds = inputData.zoomBounds;
+        if (!bounds) {
+          bounds = this._templateVars[j].tempExtentBounds;
+          zoomMax = this._templateVars[j].extentZoomBounds.maxZoom;
+          zoomMin = this._templateVars[j].extentZoomBounds.minZoom;
+          maxNativeZoom = this._templateVars[j].extentZoomBounds.maxNativeZoom;
+          minNativeZoom = this._templateVars[j].extentZoomBounds.minNativeZoom;
+        } else {
+          bounds.extend(this._templateVars[j].tempExtentBounds.min);
+          bounds.extend(this._templateVars[j].tempExtentBounds.max);
+          zoomMax = Math.max(
+            zoomMax,
+            this._templateVars[j].extentZoomBounds.maxZoom
+          );
+          zoomMin = Math.min(
+            zoomMin,
+            this._templateVars[j].extentZoomBounds.minZoom
+          );
+          maxNativeZoom = Math.max(
+            maxNativeZoom,
+            this._templateVars[j].extentZoomBounds.maxNativeZoom
+          );
+          minNativeZoom = Math.min(
+            minNativeZoom,
+            this._templateVars[j].extentZoomBounds.minNativeZoom
+          );
+        }
+      }
+    }
+    // cannot be named as layerBounds if we decide to keep the debugoverlay logic
+    this._templatedLayer.bounds = bounds;
```

Comment:
I was wondering why the _calculateBounds method should set the bounds on the _templatedLayer, and not on map-extent?

Reply:
I think it is just because in our previous discussion bounds work as a replacement/alias of **_layerBounds_**, and map-extent should not have any layerbounds property because it is an element not a layer. But it should also work, and even makes more sense if we want to bind it as a property of map-extent, just rename it as _**.extent**_ or something like this, to be consistent with what we had for layer- element.